src/Google.Api.Gax/IPageRequest.cs

```diff
@@ -0,0 +1,15 @@
+﻿/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+namespace Google.Api.Gax
+{
+    public interface IPageRequest
```

Comment:
I'd be tempted to put all the interfaces in a single file, e.g. PageStreamingInterfaces.
It's much easier to understand the whole design when they're all together.